python/GafferUI/SpreadsheetUI.py

```diff
@@ -0,0 +1,1560 @@
+
+		self.__statusLabel.setText( widget.getToolTip() )
+
+	def __leaveToolTippedWidget( self, widget ) :
+
+		self.__statusLabel.setText( "" )
+
+	def __cellsMouseMove( self, widget, event ) :
+
+		status = ""
+
+		plug = widget.plugAt( event.line.p0 )
+		if plug is not None :
+
+			rowPlug = plug.ancestor( Gaffer.Spreadsheet.RowPlug )
+			if rowPlug.getName() == "default" :
+				rowName = "Default"
+			else :
+				with self.getContext() :
+					rowName = rowPlug["name"].getValue() or "unnamed"
+
+			status = "Row : {}, Column : {}".format(
+				rowName,
+				IECore.CamelCase.toSpaced( plug.getName() ),
+			)
+
+		self.__statusLabel.setText( status )
+
+	def __cellsLeave( self, widget ) :
+
+		self.__statusLabel.setText( "" )
+
+	def __updateRowNamesWidth( self ) :
+
+		width = Gaffer.Metadata.value( self.getPlug()["default"], "spreadsheet:rowNameWidth" )
+		self.__rowNamesTable._qtWidget().setFixedWidth( width )
+
+	def __plugMetadataChanged( self, nodeTypeId, plugPath, key, plug ) :
+
+		if plug is not None and key == "spreadsheet:rowNameWidth" :
+			if self.getPlug().isAncestorOf( plug ) :
+				self.__updateRowNamesWidth()
+
+GafferUI.PlugValueWidget.registerType( Gaffer.Spreadsheet.RowsPlug, _RowsPlugValueWidget )
+
+# _PlugTableView
+# ==============
+
+class _PlugTableView( GafferUI.Widget ) :
+
+	Mode = IECore.Enum.create( "RowNames", "Defaults", "Cells" )
+
+	def __init__( self, plug, context, mode, **kw ) :
+
+		tableView = _TableView()
+		GafferUI.Widget.__init__( self, tableView, **kw )
+
+		tableView.setModel(
+			_PlugTableModel( plug, context, mode )
+		)
+
+		# Headers and column sizing
+
+		QtCompat.setSectionResizeMode( tableView.verticalHeader(), QtWidgets.QHeaderView.Fixed )
+		tableView.verticalHeader().setDefaultSectionSize( 25 )
+		tableView.verticalHeader().setVisible( False )
+
+		self.__horizontalHeader = GafferUI.Widget( QtWidgets.QHeaderView( QtCore.Qt.Horizontal, tableView ) )
+		self.__horizontalHeader._qtWidget().setDefaultAlignment( QtCore.Qt.AlignLeft )
+		tableView.setHorizontalHeader( self.__horizontalHeader._qtWidget() )
+		self.__horizontalHeader.buttonPressSignal().connect( Gaffer.WeakMethod( self.__headerButtonPress ), scoped = False )
+
+		if mode in ( self.Mode.Cells, self.Mode.Defaults ) :
+
+			for index, plug in enumerate( plug["default"]["cells"] ) :
+				width = Gaffer.Metadata.value( plug, "spreadsheet:columnWidth" )
+				if width is not None :
+					tableView.horizontalHeader().resizeSection( index, width )
+				visualIndex = Gaffer.Metadata.value( plug, "spreadsheet:columnIndex" )
+
+			self.__applySectionOrderMetadata()
+
+			tableView.horizontalHeader().setSectionsMovable( True )
+			tableView.horizontalHeader().sectionResized.connect( Gaffer.WeakMethod( self.__sectionResized ) )
+			tableView.horizontalHeader().sectionMoved.connect( Gaffer.WeakMethod( self.__sectionMoved ) )
+
+			self.__callingResizeSection = False
+			self.__callingMoveSection = False
```

Comment:
Think `__callingMoveSection` is either redundant, or needs to be set in `__applySectionOrderMetadata`?

Reply:
Good catch - e3da5247a.